linode/manager
# upcoming: [DI-21694] - Added the Create Alert Button and few components for the Create Alert Definition Form

## Description 📝
 - Added Create Alert Button alongside the tabs under Alerts. 
 - Added few components for the Create Alert Definition form
 - Added types and interfaces relevant to the Create Alert Definition workflow

## Changes  🔄
- Added the Create Alert Button along the RecentActivity, Definitions, NotificationChannel tabs under Alerts in Monitor section
- Added a BreadCrumb component in the form 
- Added Name, Description, Severity components in the form
- Added the relevant types, hooks needed for the Create Alert form workflow
- Added a validation schema for the form

## Target release date 🗓️

Please specify a release date (and environment, if applicable) to guarantee timely review of this PR. If exact date is not known, please approximate and update it as needed.

## Preview 📷

**Include a screenshot or screen recording of the change.**

:lock: Use the [Mask Sensitive Data](https://cloud.linode.com/profile/settings) setting for security.

:bulb: Use `<video src="" />` tag when including recordings in table.

| Before  | After   |
| ------- | ------- |
| ![Screenshot 2024-10-08 at 18 28 48](https://github.com/user-attachments/assets/e2a2ffab-5de6-4210-bc48-ffec739e7236) | ![Screenshot 2024-11-13 at 19 17 49](https://github.com/user-attachments/assets/a1e64a6f-3c7b-4f3b-9809-12f000f2e7e3) ![Screenshot 2024-11-13 at 19 17 38](https://github.com/user-attachments/assets/b3a89732-53ab-4cb3-b9a1-e7a8edc46704) |

## How to test 🧪

### Prerequisites
- The tabs are controlled by a featureFlag called aclpAlerting, the flag has been currently disabled. For testing enable the definitions part of the aclpAlerting flag to be true.
- The API endpoints are not live yet, so use Legacy MSW Handlers for the mock responses.

### Verification steps
- In Monitor (once the prerequisites are followed) , Alerts tab should be visible next to Dashboards.
- Under that,  Definition tab and a Create button should be visible.
- On clicking Create, the Form Page should be visible.

## As an Author, I have considered 🤔

- 👀 Doing a self review
- ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- 🤏 Splitting feature into small PRs
- ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- 🧪 Providing/improving test coverage
- 🔐 Removing all sensitive information from the code and PR description
- 🚩 Using a feature flag to protect the release
- 👣 Providing comprehensive reproduction steps
- 📑 Providing or updating our documentation
- 🕛 Scheduling a pair reviewing session
- 📱 Providing mobile support
- ♿  Providing accessibility support

<br/>


## As an Author, before moving this PR from Draft to Open, I confirmed ✅
